op-node/rollup/derive/engine_controller.go

```diff
@@ -468,3 +484,15 @@ func (e *EngineController) TryBackupUnsafeReorg(ctx context.Context) (bool, erro
 func (e *EngineController) ResetBuildingState() {
 	e.resetBuildingState()
 }
+
+func (e *EngineController) GetCurrentUnsafeHead(ctx context.Context) eth.L2BlockRef {
+	currentL2UnsafeHead := e.UnsafeL2Head()
+	if currentL2UnsafeHead.Number == 0 {
+		//derivation stage not finished yet
+		engineUnsafeHead, err := e.engine.L2BlockRefByLabel(ctx, eth.Unsafe)
+		if err == nil {
```

Comment:
I think we should not lose any err. When err is not equal to nil, we should print it to the log. If the err is not important, it can be printed as a warning.

Reply:
good tip, updated